cachito/workers/pkg_managers/rubygems.py

```diff
@@ -312,11 +322,114 @@ def _download_git_package(gem, rubygems_deps_dir, rubygems_raw_repo_name, nexus_
         # Copy downloaded archive to expected download path
         shutil.copy(repo_name.sources_dir.archive_path, download_path)
 
+    url = gem.source
+    ref = gem.version.lower()
+
     return {
-        "package": gem.name,
+        "name": gem.name,
+        "version": f"git+{url}@{ref}",
         "path": download_path,
-        "url": gem.source,
-        "ref": gem.version.lower(),
         "raw_component_name": raw_component_name,
         "have_raw_component": have_raw_component,
     }
+
+
+def _get_path_package_info(dep, bundle_dir, package_root):
+    """
+    Get info about PATH dependency including path relative to the bundle source root.
+
+    :param GemMetadata dep: path dependency
+    :param bundle_dir: the root of the bundle with app source code
+    :param package_root: path to the root of the processed package
+    :return: dict with name, version and path keys
+    """
+    path = Path(package_root / dep.source).resolve().relative_to(bundle_dir.source_root_dir)
+
+    return {
+        "name": dep.name,
+        "version": dep.version,
+        "path": path,
+    }
```

Comment:
What will be the value of `dep.version` here? The path to the dependency or a version similar to those of GEM deps? Note that, when generating the content manifest later, we will need to be able to tell the difference between a PATH dependency and a GEM dependency.

Reply:
Even for PATH dependencies, [version is resolved](https://github.com/3scale/porta/blob/ce87a98f0f93696155dd9602220e5093bc278386/Gemfile.lock#L80) (based on .gemspec of that dependency). So here the version could be "1.0.0". Cachito doesn't really need that version later, so one way of telling the difference would be setting the version to "PATH" and then during content manifest generation considering all gems with `version == "PATH"` to be path dependencies. The second, and IMO a cleaner option is to add `kind` slot to `Package` in `content_manifest.py`, as you can see in #684.

Reply:
+1 for `kind` slot to `Package` in `content_manifest.py` for better visibility

Reply:
You should also consider how such dependencies will be presented in the request JSON https://github.com/containerbuildsystem/cachito/blob/master/docs/metadata.md#request-json

I would avoid using versions like "1.0.0" for PATH deps, as that could lead to confusion where the consumers of Cachito metadata would think that the dependency comes from the rubygems registry.

Reply:
Oh, I see. I don't really understand the whole workflow of generating request JSON, but I assume it's produced from the same JSON file that content manifest generation is using. 

Therefore, to match the rest of the package managers, `resolve_rubygems` could return for each dependency `name`, `type`, `version` (which for PATH dependencies will contain just the path, ad gomod does, and for GIT it could be same as for pip `git+url@hash`, so that it's clear it's a different type of dependency) and maybe `dev`. `replaces` don't have to be returned, because it is filled with null automatically, if nothing is specified.

Now the question is, do we want to return `kind` as well (and possibly the other metadata that are returned in the current implementation)? I think there's no harm in doing that, and it will be better than [what `gomod` is doing](https://github.com/containerbuildsystem/cachito/blob/28a6afc130df5296945898b7d8578e5bf2157ac0/cachito/web/content_manifest.py#L322), just assuming that every version starting with `.` is a PATH dependency. What do you think?

Reply:
> Oh, I see. I don't really understand the whole workflow of generating request JSON, but I assume it's produced from the same JSON file that content manifest generation is using.

Yeah, it's the same file. The request JSON is generated here (the packages and dependencies sections, specifically): https://github.com/containerbuildsystem/cachito/blob/6c3de13b9b72f243218467511f04d04b96c8cbd3/cachito/web/models.py#L457-L459

AFAIK this pretty much just dumps the content of the JSON directly into the API response.

> Therefore, to match the rest of the package managers, resolve_rubygems could return for each dependency name, type, version (which for PATH dependencies will contain just the path, ad gomod does, and for GIT it could be same as for pip git+url@hash, so that it's clear it's a different type of dependency) and maybe dev. replaces don't have to be returned, because it is filled with null automatically, if nothing is specified.

+1 for the versions. Returning `dev` is not necessary if we don't have that info

> Now the question is, do we want to return kind as well (and possibly the other metadata that are returned in the current implementation)?

Ideally, the metadata returned in the API response should be common for all package managers. A `kind` key used only for ruby would not be ideal, but if it's documented well it would be OK. Another thing to consider would be that, in case we eventually want to add `kind` for other package managers, there should be a common set of values for the `kind` key.

That said, it would also be OK to  save `kind` in packages.json but clean it out of the API response

Reply:
> That said, it would also be OK to save kind in packages.json but clean it out of the API response

(but PATH deps should still be distinguishable from GEM deps)